djongo/sql2mongo/operators.py

```diff
@@ -110,7 +110,8 @@ def to_mongo(self):
         raise NotImplementedError
 
     def _to_mongo(self, op):
-        if self.query.nested_query is not None:
+        ## FIX: incorrectly resolving to $_nested_in for all ops instead of the relevant one only.
+        if self.query.nested_query is not None and not self._in:
```

Comment:
what happens when you something like 
where abc in (select ...) ?

Reply:
that will resolve to {'$expr': {'$in': ['$abc', '$_nested_in']}}..

In the problematic sql, this
```
 {'$match': {'$or': [{'$and': [{'$expr': {'$in': ['$trial', '$_nested_in']}}, 
                               {'$expr': {'$in': ['$id', '$_nested_in']}}]},
                     {'$and': [{'$expr': {'$in': ['$trial', '$_nested_in']}}, 
                               {'id': None}]}]}},
```
becomes
```
 {'$match': {'$or': [{'$and': [{'$expr': {'$in': ['$trial', ['testset']]}}, 
                               {'$expr': {'$in': ['$id', '$_nested_in']}}]},
                     {'$and': [{'$expr': {'$in': ['$trial', ['testset']]}},
                               {'id': None}]}]}},
```

Reply:
Perfect